extended/src/main/java/apoc/util/ExtendedUtil.java

```diff
@@ -43,4 +66,129 @@ public static String toCypherMap( Map<String, Object> map) {
+            case "Short":
+                return Short.parseShort(value);
+            case "Int":
+                return Integer.parseInt(value);
+            case "Long":
+                return Long.parseLong(value);
+            case "Node", "Relationship":
+                return JsonUtil.parse(value, null, Map.class);
+            case "NO_VALUE":
+                return null;
+            default:
+                // If ends with "Array", for example StringArray
+                if (typeName.endsWith("Array")) {
+                    value = StringUtils.removeStart(value, "[");
+                    value = StringUtils.removeEnd(value, "]");
+                    String array = typeName.replace("Array", "");
+
+                    final Object[] prototype = getPrototypeFor(array);
+                    return Arrays.stream(value.split(","))
+                            .map(item -> convertValue(StringUtils.trim(item), array))
+                            .toList()
+                            .toArray(prototype);
+                }
+                return value;
+        }
+    }
+
+    private static PointValue getPointValue(String value) {
+        try {
+            return PointValue.parse(value);
+        } catch (Neo4jException e) {
+            // fallback in case of double-quotes, e.g. {"crs":"wgs-84-3d","latitude":13.1,"longitude":33.46789,"height":100.0}
+            // we remove the double quotes before parsing the result, e.g. {crs:"wgs-84-3d",latitude:13.1,longitude:33.46789,height:100.0}
+            ObjectMapper objectMapper = new ObjectMapper()
+                    .disable(JsonWriteFeature.QUOTE_FIELD_NAMES.mappedFeature());
+            try {
+                Map readValue = objectMapper.readValue(value, Map.class);
+                String stringWithoutKeyQuotes = objectMapper.writeValueAsString(readValue);
+                return PointValue.parse(stringWithoutKeyQuotes);
+            } catch (JsonProcessingException ex) {
+                throw new RuntimeException(ex);
+            }
+        }
+    }
+
+    // similar to CsvPropertyConverter
+    public static Object[] getPrototypeFor(String type) {
+        return switch (type) {
+            case "Long" -> new Long[]{};
+            case "Integer" -> new Integer[]{};
+            case "Double" -> new Double[]{};
+            case "Float" -> new Float[]{};
+            case "Boolean" -> new Boolean[]{};
+            case "Byte" -> new Byte[]{};
+            case "Short" -> new Short[]{};
+            case "Char" -> new Character[]{};
+            case "String" -> new String[]{};
+            case "DateTime" -> new ZonedDateTime[]{};
+            case "LocalTime" -> new LocalTime[]{};
+            case "LocalDateTime" -> new LocalDateTime[]{};
+            case "Point" -> new PointValue[]{};
+            case "Time" -> new OffsetTime[]{};
+            case "Date" -> new LocalDate[]{};
+            case "Duration" -> new DurationValue[]{};
```

Comment:
maybe we could add:
```suggestion
            case "Map" -> new Object[]{};
```

Non so se può esistere uno scenario che lo richiede, verifica. In caso non esista ignora il commento

Reply:
Penso di no, considerando che non possiamo creare nodi con proprietà `Map` / `List of Map`

Reply:
Sicuramenten non possiamo tornare nodi con quelle proprietà però non so se permettiamo di esportare query custom, in quel caso dobbiamo gestirlo.

Reply:
Se non ho capito male, sì possiamo esportare delle query custom tipo:
```
CALL apoc.export.arrow.query('query_test.arrow',  [{a: <foo>, b: <bar>}] AS mapList) YIELD file
```

ma, consistentemente con le altre procedure, comunque non possiamo utilizzare la apoc.import.arrow con un file creato con la export query, 
ma dobbiamo utilizzare la apoc.load che non viene gestita in questo punto, ma si trova in apoc core.

Reply:
Secondo me va gestito qui la export. E poi la load